exa/core/box.py

```diff
@@ -31,30 +45,96 @@ def groupby(self):
         pass
 
     def info(self):
-        pass
+        info = []
+        for name, data in self._data.items():
+            df = data.data()
+            # TODO: prepend __module__?
+            if isinstance(df, pd.DataFrame):
+                info.append((name, type(data), df.memory_usage().sum(), df.shape))
+            elif isinstance(df, pd.Series):
+                info.append((name, type(data), df.memory_usage(), df.shape))
+            else:
+                info.append((name, type(data), None, None))
+        info = pd.DataFrame(info, columns=['name', 'type', 'size', 'shape'])
+        return info.set_index('name').sort_index()
 
     def memory(self):
         pass
 
     def network(self):
-        pass
+        """Build a network of the relationships between
+        data as specified by the container's metadata.
+        """
+        # TODO: log-scale sizes for nodes
+        sizes = self.info()['size'].to_dict()
+        # TODO: there is a bug in this
+        related = {}
+        print("name0, name1, index0, data1.indexes, data1.columns")
+        for name0, data0 in self._data.items():
+            for name1, data1 in self._data.items():
+                if data0 is data1:
+                    continue
+                for index0 in data0.indexes:
+                    print(name0, name1, index0, data1.indexes, data1.columns)
+                    if index0 in data1.indexes:
+                        related[(name0, name1)] = 'index-index'
+                        related[(name1, name0)] = 'index-index'
+                    for col1 in data1.columns:
+                        print(col1, index0, col1.startswith(index0))
+                        if col1.startswith(index0):
+                            related[(name0, name1)] = 'index-column'
+                            related[(name1, name0)] = 'column-index'
+                            print(related)
+        g = nx.Graph()
+        g.add_nodes_from(sizes.keys())
+        g.add_edges_from(related.keys())
+        g.edge_types = related
+        return g
 
-    def save(self, name=None, directory=None, parquet_backend='pyarrow'):
-        """Save a bundle of files to a tarball."""
+    def save(self, name=None, directory=None):
+        """Save a bundle of files to a tarball. Not sure
+        how parquet will interact with tarfile buffer so
+        taking the easy way out for now."""
+        # TODO : box needs to additionally save its
+        #        own metadata like Data
         name = name or self.hexuid
-        directory = Path(directory) or exa.cfg.savedir
-        bundle = directory / name
+        adir = exa.cfg.savedir
+        if directory is not None:
+            adir = Path(directory)
+        bundle = adir / name
```

Comment:
We could `.as_posix` here and then on line 108 do `tarfl = bundle + '.tar.gz'

Reply:
At this point the `save` method on the `Data` requires a `Path` object, but you are right we can just `bundle.as_posix()` on line 108 anyway.

Reply:
Oh good point. I missed that!